lightning_pose/utils/predictions.py

```diff
@@ -128,21 +130,22 @@ def unpack_preds(
 
         return stacked_preds, stacked_confs
 
-    def fix_context_preds_confs(self, stacked_preds: TensorType, is_confidence: bool = False):
+    @staticmethod
+    def fix_context_preds_confs(stacked_preds: TensorType, is_confidence: bool = False):
         """
         In the context model, ind=0 is associated with image[2], and ind=1 is associated with image[3],
         so we need to shift the predictions and confidences by two and eliminate the edges.
         NOTE: confidences are not zero in the first and last two images, they are instead replicas of images[-2] and images[-3]
         """
         # first pad the first two rows for which we have no valid preds.
-        preds_1 = torch.tile(stacked_preds[0], (2,1)) # copying twice the prediction for image[2]
-        preds_2 = stacked_preds[0:-2] # throw out the last two rows.
+        preds_1 = torch.tile(stacked_preds[0], (2, 1))  # copying twice the prediction for image[2]
+        preds_2 = stacked_preds[0:-2]  # throw out the last two rows.
         preds_combined = torch.vstack([preds_1, preds_2])
         # after concat this has the same length. everything is shifted by two rows.
         # but we don't have valid predictions for the last two elements, so we pad with element -3.
         preds_combined[-2:, :] = preds_combined[-3, :]
 
-        if is_confidence == True:
+        if is_confidence:
```

Comment:
for the `mhcrnn` we don't have to do this, since we are also forming prediction with the single-frame head. agree? this `if` just applies for the simple context model.